### PR TITLE
Refactor README: Add Algorithms, Reorganize Sections & Fix Documentation

### PR DESCRIPTION
**Pull request by OzPol**
# Description

- Added new algorithms: A*, Kruskal's, Bellman-Ford, Prim's, Gale-Shapley, Topological Sort, Hungarian Algorithm, AVL Tree, Red-- - Black Tree, and Ford-Fulkerson.
- Reorganized the algorithms and data structures sections alphabetically for better navigation.
- Updated placeholder links with active documentation references.
- Fixed incorrect logos for unimplemented algorithms (Python and other language logos replaced with GitHub placeholders).
- Addressed lint issues: fixed broken links and removed unnecessary tabs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1283,7 +1283,7 @@ In order to achieve greater coverage and encourage more people to contribute to
             </td>
         </tr>
         <tr>
-            <td>Min and Max (D&C)</td>
+            <td><a href="https://www.tutorialspoint.com/data_structures_algorithms/max_min_problem.htm">Min and Max (D&C)</a></td>
```
Comment (ricardocarva) [inline README.md]: Good catch

Comment (ricardocarva) [reply README.md]: For all of these

File: README.md [hunk 2]
```diff
@@ -4089,7 +4089,7 @@ In order to achieve greater coverage and encourage more people to contribute to
             </td>
             <td> <!-- Python -->
                 <a href="./CONTRIBUTING.md">
-                    <img align="center" height="25" src="./logos/python.svg" />
+                    <img align="center" height="25" src="./logos/github.svg" />
```
Comment (ricardocarva) [inline README.md:4092]: For a brief moment based on the PR description, I thought there were existing algorithms that were missing the logos until seeing the commit referring to that.

I sounds like you meant the 10 new algorithms you included in the commit with hash 43363dc52b1886fbcfad88692c489de5342f1117 were missing the logos and the comment was addressing that.

Comment (ricardocarva) [reply README.md:4092]: Is that correct, @OzPol?

File: README.md (hunk 1 above)
Comment (ricardocarva) [reply README.md]: Thanks for fixing all of these.

Comment (ricardocarva) [reply README.md]: Not sure why I cant resolve this one.

File: README.md (hunk 2 above)
Comment (ricardocarva) [reply README.md:4092]: Thank you for the contribution, @OzPol. I checked the code, and overall it looks great.
I think once you confirm this question, we should be good to approve/merge this.

Comment (OzPol) [reply README.md:4092]: Hi @ricardocarva,

Thank you for your review and follow-up! Yes, you're correct – the comment regarding the missing logos pertains to the 10 new algorithms added in commit 43363dc. Specifically, I mistakenly copied the Python implementation placeholder from an existing algorithm for some entries (like the Red-Black Tree and Bellman-Ford). I corrected this in a follow-up commit by replacing the incorrect logos with GitHub placeholders to ensure consistency for unimplemented algorithms.

However, before submitting the PR, I noticed a broken link in the README, unrelated to the new additions. I fixed that as well, which initially caused a lint check failure but was resolved afterward.

Please let me know if anything else needs adjustment!
Thanks again for your review and support.

**Review by kelvins — APPROVED** [review]
Nice, it looks good to me! Thanks a lot! 👏

**Pull request merged**